tests/functional/base_pages/list-collector-summary.page.js

```diff
@@ -0,0 +1,43 @@
+const BasePage = require('./base.page');
+
+class ListCollectorSummary extends BasePage {
+
+  listCollectorPeopleRowAdd() {
+    return '[data-qa=people-list-summary] [data-qa=add-item-link]';
+  }
+
+  listCollectorPeopleRowChange(number = 1) {
+    return '[data-qa=people-list-summary] tbody:nth-child(' + number + ') [data-qa=change-item-link]';
+  }
+
+  listCollectorPeopleRowRemove(number = 1) {
+    return '[data-qa=people-list-summary] tbody:nth-child(' + number + ') [data-qa=remove-item-link]';
+  }
+
+  listCollectorPeopleRowTitle(number = 1) {
+    return '[data-qa=people-list-summary] tbody:nth-child(' + number + ') tr td.summary__item-title';
+  }
+
+  listCollectorVisitorRowAdd() {
+    return '[data-qa=visitors-list-summary] [data-qa=add-item-link]';
+  }
+
+  listCollectorVisitorRowChange(number = 1) {
+    return '[data-qa=visitors-list-summary] tbody:nth-child(' + number + ') [data-qa=change-item-link]';
+  }
+
+  listCollectorVisitorRowRemove(number = 1) {
+    return '[data-qa=visitors-list-summary] tbody:nth-child(' + number + ') [data-qa=remove-item-link]';
+  }
+
+  listCollectorVisitorRowTitle(number = 1) {
+    return '[data-qa=visitors-list-summary] tbody:nth-child(' + number + ') tr td.summary__item-title';
+  }
+
+  listLabel(instance) { return `tbody:nth-child(${instance}) td:first-child`; }
+
+  peopleListLabel(listItemInstance) { return 'div[data-qa="people-list-summary"] tbody:nth-child(' + listItemInstance + ') td:first-child'; }
+
+}
+
+module.exports = new ListCollectorSummary();
```

Comment:
There is code in `generate_pages.py` that handles creating page specs from `SectionSummary` definitions. This needs to be refactored into logic that generates an appropriate page object based on the contents of the section summary property or the answers in the section.

This is important as currently:
- This will not work for list summaries containing something other than `people` or `visitors`.
- References to answers have changed from things like `PersonalSummaryPage.dateOfBirthQuestion()` to `SectionSummaryPage.summaryRowTitle(2)`, which is far less readable.